### ISSUE TITLE
Useless/confusing "go straight" on a very short road

### ISSUE THREAD

**Issue by RicoElectrico** [organicmaps/organicmaps#6031]
**Describe the issue**
On the route below the instructions are:
- Turn left, then go straight
- Turn right

**Steps to reproduce**
Take this route: https://www.openstreetmap.org/directions?engine=fossgis_valhalla_car&route=54.56072%2C18.50978%3B54.56152%2C18.50867#map=19/54.56122/18.50979

**Expected behavior**
No "go straight" direction is spoken.
Actually, is it expected that "X, then Y" is announced, but "Y" is not announced on its own?

**Screenshots**
![obraz](https://github.com/organicmaps/organicmaps/assets/1409116/32069a1f-6c33-40ab-b9d7-99532d2a6d4d)

**Comment by RicoElectrico** [comment]
Oh, I see on the desktop with debug turns display:
![obraz](https://github.com/organicmaps/organicmaps/assets/1409116/dbdde334-b20f-4c94-af1a-b087ed4f89e2)
It appears that junctions trigger this.

**Comment by biodranik** [comment]
Good catch! Let's find a proper way to filter these unnecessary announcements.

**Comment by RicoElectrico** [comment]
It's an exception, usually on such intersections OM doesn't announce "go straight". Here maybe the angle of the road is causing this.

I think there is a chance of confusion resulting in going into _Kapitana Antoniego Ledóchowskiego_. Which is quite bad in this case, there is no path to the destination other than calculated by OM (turning right here leads to a no-exit section) and it would mean wasting a lot of time to turn around.
